### DISCUSSION TITLE
How to make eglot full fledge IDE?

### DISCUSSION THREAD

**Discussion by 99702** [joaotavora/eglot#1144] [Q&A]
Hello, 

I'm using eglot from emacs 29 branch. Before eglot i used to use lsp-mode. But i cant see any 
code completions, linter, error highlighter, automatic imports. 
What are packages and eglot configurations would you like me to add if i want full fledge ide  just like lsp-mode or vscode?

Thank you for your help

**Comment by chriselrod** [comment]
Anything that works with completion-at-point-functions will work for completions.
`C-M-i` is builtin in, but I've been using company, but there are other options like corfu.
Things like go to definition are built in https://www.gnu.org/software/emacs/manual/html_node/emacs/Looking-Up-Identifiers.html

See ElDoc and Flymake for documentation and diagnostics, respectively.
